gc.c

```diff
@@ -3443,6 +3496,8 @@ obj_free(rb_objspace_t *objspace, VALUE obj)
         break;
     }
 
+    RVALUE_AGE_RESET(obj);
```

Comment:
`heap_page_add_freeobj` also resets the age. Do we need to do it in both places?